orchagent/aclorch.cpp

```diff
@@ -981,6 +981,73 @@ void AclTable::link(sai_object_id_t portOid)
     ports.emplace(portOid, SAI_NULL_OBJECT_ID);
 }
 
+bool AclTable::add(shared_ptr<AclRule> newRule)
+{
+    string rule_id = newRule->getId();
+    auto ruleIter = rules.find(rule_id);
+    if (ruleIter != rules.end())
+    {
+        // If ACL rule already exists, delete it first
+        if (ruleIter->second->remove())
+        {
+            rules.erase(ruleIter);
+            SWSS_LOG_NOTICE("Successfully deleted ACL rule: %s", rule_id.c_str());
+        }
+    }
+
+    if (newRule->create())
+    {
+        rules[rule_id] = newRule;
+        SWSS_LOG_NOTICE("Successfully created ACL rule %s in table %s", rule_id.c_str(), id.c_str());
+        return true;
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Failed to create rule in table %s", id.c_str());
+        return false;
```

Comment:
Since you are doing the refactoring,  I'm curious to see how you handle the error return code all the levels up the stack?  I have similar issues in other PR and would like to get some hint here.

Reply:
If you are talking specific to this function, the call stacks is like below.
```
bool AclTable::add(shared_ptr<AclRule> newRule)
bool AclOrch::addAclRule(shared_ptr<AclRule> newRule, string table_id)
void AclOrch::doAclRuleTask(Consumer &consumer)
```
You can see if any lower-level function is possible to fail, it returns a false result, and upper-level function should check the result and response to the failure. Upper-level function should never ignore the returned result. That is the point when I asked you to 'handle the error'.

Reply:
What happens in AclTable::add() upon error?   Just log the error without quit or recover?

Reply:
Yes, you are right. Base on current implementation, this is the best we could do.

Reply:
Understood